include/zephyr/drivers/i3c.h

```diff
@@ -0,0 +1,2443 @@
+/*
+ * Copyright 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_I3C_H_
+#define ZEPHYR_INCLUDE_DRIVERS_I3C_H_
+
+/**
+ * @brief I3C Interface
+ * @defgroup i3c_interface I3C Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#include <zephyr/drivers/i3c/addresses.h>
+#include <zephyr/drivers/i3c/ccc.h>
+#include <zephyr/drivers/i3c/ibi.h>
+#include <zephyr/drivers/i2c.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
```

Comment:
Is BCR something the user would set?

Reply:
No. BCR comes from the device, and downstream drivers may need this (for example, different variants of a sensor which may have different capabilities, or that the same sensor but with different hardware configurations).